cpp/src/interpolate/cubic_spline.cu

```diff
@@ -0,0 +1,409 @@
+    TPRINT(i_buffer, "i_i");
+    TPRINT(D_buffer, "D_i");
+    TPRINT(Dlu_buffer, "Dlu_i");
+    TPRINT(u_buffer, "u_i");
+
+    // cusparse solve n length m tridiagonal systems
+    // 4. call cusparse<T>gtsv2() to solve
+    // 4.1 Get cuSparse library context
+    // compute inputs:
+    // handle: the cuSparse library context
+    // m: size
+    // n: number of columns of solution matrix B
+    // dl, d, du: vectors of the diagonal
+    // B: (ldb, n) dimensional dense matrix to be solved for
+    // ldb: leading dimension of B
+    // pBuffer: get size of thisu by gtsv2_bufferSizeExt 
+    cusparseStatus_t cusparseStatus;
+    cusparseHandle_t handle;
+    cudaMalloc(&handle, sizeof(cusparseHandle_t));
+    cusparseStatus = cusparseCreate(&handle);
+    detail::HANDLE_CUSPARSE_STATUS(cusparseStatus);
+    size_t pBufferSize;
+    cusparseStatus = cusparseSgtsv2StridedBatch_bufferSizeExt(
+        handle,
+        y.size()/(prefixes.size()-1),
+        Dll_buffer.data<float>(),
+        D_buffer.data<float>(),
+        Dlu_buffer.data<float>(),
+        u_buffer.data<float>(),
+        prefixes.size()-1,
+        y.size()/(prefixes.size()-1),
+        &pBufferSize
+    );
+    detail::HANDLE_CUSPARSE_STATUS(cusparseStatus);
+    rmm::device_vector<float> pBuffer(pBufferSize);
+    cusparseStatus = cusparseSgtsv2StridedBatch(
+        handle,
+        y.size()/(prefixes.size()-1),
+        Dll_buffer.data<float>(),
+        D_buffer.data<float>(),
+        Dlu_buffer.data<float>(),
+        u_buffer.data<float>(),
+        prefixes.size()-1,
+        y.size()/(prefixes.size()-1),
+        thrust::raw_pointer_cast(pBuffer.data())
+    );
+    detail::HANDLE_CUSPARSE_STATUS(cusparseStatus);
+    cusparseStatus = cusparseDestroy(handle);
+    detail::HANDLE_CUSPARSE_STATUS(cusparseStatus);
+
+    int dn = n - (prefixes.size()-1);
+    // Finally, compute coefficients via Horner's scheme
+    auto d3_col = make_numeric_column(y.type(), dn, cudf::mask_state::UNALLOCATED, stream, mr);
+    auto d2_col = make_numeric_column(y.type(), dn, cudf::mask_state::UNALLOCATED, stream, mr);
+    auto d1_col = make_numeric_column(y.type(), dn, cudf::mask_state::UNALLOCATED, stream, mr);
+    auto d0_col = make_numeric_column(y.type(), dn, cudf::mask_state::UNALLOCATED, stream, mr);
+    cudf::mutable_column_view d3 = d3_col->mutable_view();
+    cudf::mutable_column_view d2 = d2_col->mutable_view();
+    cudf::mutable_column_view d1 = d1_col->mutable_view();
+    cudf::mutable_column_view d0 = d0_col->mutable_view();
+
+    coefficients_compute coefs;
+    coefs.operator()<float>(t, y, prefixes, h_buffer, i_buffer, u_buffer, d3, d2, d1, d0, mr, stream);
+
+    TPRINT(h_buffer, "h_buffer_");
+    TPRINT(i_buffer, "i_buffer_");
+    TPRINT(u_buffer, "u_buffer_");
+
+    TPRINT(d3, "d3");
+    TPRINT(d2, "d2");
+    TPRINT(d1, "d1");
+    TPRINT(d0, "d0");
+    
+    // Place d3..0 into a table and return
+    std::vector<std::unique_ptr<cudf::column>> table;
```

Comment:
The farthest I can get this to work is 
```
    auto result = std::make_unique<cudf::experimental::table>({
        std::move(static_cast<cudf::column>(*d3_col)),
        std::move(static_cast<cudf::column>(*d2_col)),
        std::move(static_cast<cudf::column>(*d1_col)),
        std::move(static_cast<cudf::column>(*d0_col))
    });
```
Which just says that I can't pass {...} in to `make_unique`

```
Scanning dependencies of target cuspatial
[  2%] Building CUDA object CMakeFiles/cuspatial.dir/src/interpolate/cubic_spline.cu.o
/home/thomson/NVIDIA/cuspatial/cpp/src/interpolate/cubic_spline.cu(405): error: no instance of overloaded function "std::make_unique" matches the argument list
            argument types are: ({...})

1 error detected in the compilation of "/tmp/tmpxft_00006be1_00000000-7_cubic_spline.compute_70.cpp1.ii".
CMakeFiles/cuspatial.dir/build.make:62: recipe for target 'CMakeFiles/cuspatial.dir/src/interpolate/cubic_spline.cu.o' failed
```

Reply:
Maybe we need a FLOAT4 type in libcudf...

Reply:
That doesn't seem necessary? I'm building a 4 column table, there's no need to modify the tabular format of cuDF because of this one cuspatial requirement, imo.